pennylane/operation.py

```diff
@@ -1039,9 +1039,11 @@ def eigvals(self):
     def __init__(self, *params, wires=None, do_queue=True, id=None):
         # extract the arguments
         if wires is None:
-            wires = params[-1]
+            try:
+                wires = params[-1]
+            except IndexError:
+                raise Exception("Can only access when arguments are passed.")
```

Comment:
Maybe it's good to be slightly more specific when informing the user. Accepting this new message will also require the test case to be adjusted with the new message.
```suggestion
                raise Exception("Cannot extract wires as no arguments were passed to the observable.")
```